### ISSUE TITLE
Set-UnifiedGroup -Identity -PrimarySmtpAddress permission issue

### ISSUE THREAD

**Issue by TonyLi2013** [MicrosoftDocs/office-docs-powershell#6923]
Hi Team,

By Get-ManagementRole command, we found the Set-UnifiedGroup need **Mail Recipients Role**:

$Perms = Get-ManagementRole -Cmdlet Set-UnifiedGroup -CmdletParameters PrimarySmtpAddress
$Perms | foreach {Get-ManagementRoleAssignment -Role $_.Name -Delegating $false | Format-Table -Auto Role,RoleAssigneeType,RoleAssigneeName}

But one user who have Mail Recipients Role failed to execute the Set-UnifiedGroup -Identity -PrimarySmtpAddress command, error message is:

We failed to update the unified group. Please try again later.
+ CategoryInfo : NotSpecified: (:) [Set-UnifiedGroup], TaskException
---

According to our research and test, for Set-UnifiedGroup command, account needs to have at least Exchange admin permission, is that correct?

#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 66f0a60b-0a31-dd92-53eb-4f7def6319f3
* Version Independent ID: 06ad3c31-d631-cb66-07e7-380f9f923ff9
* Content: [Find the permissions required to run any Exchange cmdlet](https://docs.microsoft.com/en-us/powershell/exchange/find-exchange-cmdlet-permissions?view=exchange-ps)
* Content Source: [exchange/docs-conceptual/find-exchange-cmdlet-permissions.md](https://github.com/MicrosoftDocs/office-docs-powershell/blob/master/exchange/docs-conceptual/find-exchange-cmdlet-permissions.md)
* Service: **exchange-powershell**
* GitHub Login: @chrisda
* Microsoft Alias: **chrisda**

**Comment by chrisda** [comment]
@TonyLi2013, I confirmed your results by creating a custom role group that only has the Mail Recipients role assigned to it, and I added a random user to the role group.

Trying to change the values of the PrimarySmtpAddress, EmailAddresses, or DisplayName parameters results in that error. Changing CustomAttribute1 is successful.

A global admin (Organization Management role group in Exchange) is able to update PrimarySmtpAddress.

There seems to be longstanding issues with updating the primary email address on M365 groups. According to [this thread](https://techcommunity.microsoft.com/t5/microsoft-365-groups/change-o365-group-email-address-using-powershell/m-p/132556/page/2), a Premier support case resulted in the revelation that only Global Admins can change the primary email address.
